src/sw.ts

```diff
@@ -101,7 +105,46 @@ self.addEventListener('fetch', (event) => {
     // intercept and do our own stuff...
     event.respondWith(fetchHandler({ path: url.pathname, request }))
   } else if (isSubdomainRequest(event)) {
-    event.respondWith(fetchHandler({ path: url.pathname, request }))
+    const isMutable = protocol === 'ipns'
+    const cacheKey = `${event.request.url}-${event.request.headers.get('Accept') ?? ''}`
+    const cacheName = isMutable ? MUTABLE_CACHE : IMMUTABLE_CACHE
+    log('helia-sw: cache name: %s | key: %s', cacheName, cacheKey)
+
+    event.respondWith((async () => {
+      const cache = await caches.open(cacheName)
+      const cachedResponse = await cache.match(cacheKey)
+
+      if ((cachedResponse != null) && !hasExpired(cachedResponse)) {
+        // If there is an entry in the cache for event.request,
+        // then response will be defined and we can just return it.
+        log('helia-ws: cached response HIT for %s (expires: %s) %o', cacheKey, cachedResponse.headers.get('Expires'), cachedResponse)
+
+        trace('helia-ws: updating cache for %s in the background', cacheKey)
+        // 👇 update cache in the background wihtout awaiting
+        fetchHandler({ path: url.pathname, request })
```

Comment:
I think we should move the cache logic into the fetchHandler

Reply:
Have a look at how I structured it now, but I think it makes sense to keep the two functions separate, where fetchHandler actually handles fetching with helia while getResponseFromCacheOrFetch handles caching and delegating to fetchHandler.

Reply:
yep, totally what I was thinking :)